skimage/color/colorconv.py

```diff
@@ -804,6 +804,46 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create an RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        alpha channel of the output image. Same shape as ``image``.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+
+    """
+
+    arr = dtype.img_as_float(image)
```

Comment:
why are you so interested in making the image a float? I don't think there is any loss of information if you leave this as the input dtype.

Reply:
I don't really know how to manage alpha channel if `image.dtype` is float and `image` values are not between 0 and 1 :-S.

Reply:
I just realized that I have to check if `0 <= alpha <= 1` :-)

Reply:
I guess I would simply not worry about that.

Junk in means junk out for me.

At the very least, if dtype is uint8, which many images are, then it is very likely OK to keep it as that.

Reply:
I am not in favor of this "junk in junk out" behavior when it can be avoided.
I understand the uint8 case, but what about the int case? The user will end up with an alpha channel filled with 9223372036854775807!
The majority of the functions of this module converts the input image to float, so I guess that this conversion will not be that surprising for the user.
But well, if a majority votes for no conversion...

Reply:
@rfezzani `int` images are a big problem for us in general. I recommend you read #3009 for some background if you haven't already. The converse problem is that converting to float will probably result in very tiny values if the image is higher depth than 16-bit.

But I very much would like to preserve the type for uint8 images at least, which is a common use case. Probably 16-bit as well...

Reply:
according to our data types document, setting the alpha channel to 2^32-1 is the right thing to do:
https://scikit-image.org/docs/dev/user_guide/data_types.html

The thing is that data validation isn't cheap, especially with image data. If you want to validate at every step, you are essentially guaranteeing that nobody that cares about performance will use these functions.

For batch image processing, everybody will start to look elsewhere if calls to `min` and `max` are performed multiple times in the library without recourse to remove them.

Reply:
I know that 2^32-1 is the correct value, I am just saying that in this case, "junk in" results in "huge :hankey: "!
:smile: 
I am not in favor of validation, but conversion is fine since it is used elsewhere in multiple functions in `skimage`.
By the way, I removed the call to `img_as_float`...